test/account_api_test.rb

```diff
@@ -275,4 +275,62 @@
       end
     end
   end
+
+  describe "fetching attribute names" do
+    describe "the user is logged in" do
+      before do
+        account_api
+          .given(given)
+          .upon_receiving("a get-attributes-names request")
+          .with(
+            method: :get,
+            path: "/api/attributes/names",
+            query: { "attributes[]" => queried_attribute_names },
+            headers: GdsApi::JsonClient.default_request_headers.merge(authenticated_headers),
+          )
+          .will_respond_with(
+            status: 200,
+            headers: { "Content-Type" => "application/json; charset=utf-8" },
+            body: {
+              govuk_account_session: Pact.like("user-session-id"),
+              values: returned_attribute_names,
+            },
+          )
+      end
+
+      let(:queried_attribute_names) { %w[foo] }
+
+      describe "attributes do not exist" do
+        let(:given) { "there is a valid user session" }
+        let(:returned_attribute_names) { [] }
+
+        it "responds with 200 OK, a new govuk_account_session, and no attributes" do
+          response = api_client.get_attributes_names(govuk_account_session: govuk_account_session, attributes: queried_attribute_names)
+          assert response["govuk_account_session"].present?
+          assert_equal returned_attribute_names, response["values"]
+          assert_equal 200, response.code
+        end
+      end
+
+      describe "attributes exist" do
+        let(:given) { "there is a valid user session, with an attribute called 'foo'" }
+        let(:returned_attribute_names) { %w[foo] }
+
+        it "responds with the attribute values" do
+          response = api_client.get_attributes_names(govuk_account_session: govuk_account_session, attributes: queried_attribute_names)
+          assert_equal returned_attribute_names, response["values"]
+        end
+      end
+
+      describe "attributes with no value exist" do
+        let(:given) { "there is a valid user session, with an attribute called 'foo' that has no value" }
+        let(:returned_attribute_names) { [] }
+
+        it "responds with the attribute values" do
+          response = api_client.get_attributes_names(govuk_account_session: govuk_account_session, attributes: queried_attribute_names)
+          assert_equal returned_attribute_names, response["values"]
+        end
+      end
```

Comment:
Is this testing anything more than the `responds with 200 OK, a new govuk_account_session, and no attributes` test?

Reply:
That test is also checking that a 404ing attribute is not included in the response

Reply:
I just closed the [related PR](https://github.com/alphagov/account-api/pull/62) that introduced the related `provider_state`. If an attribute isn't present or it has a `nil` value, the response would be the same, and we don't want attributes with `nil` values anyway (see [this comment](https://github.com/alphagov/account-api/pull/62#issuecomment-833427248)).